cli/internal/hashing/package_deps_hash.go

```diff
@@ -371,10 +380,13 @@ func gitHashObject(anchor turbopath.AbsoluteSystemPath, filesToHash []turbopath.
 	return output, nil
 }
 
-func manuallyHashFiles(rootPath turbopath.AbsoluteSystemPath, files []turbopath.AnchoredSystemPath) (map[turbopath.AnchoredUnixPath]string, error) {
-	hashObject := make(map[turbopath.AnchoredUnixPath]string)
+func manuallyHashFiles(rootPath turbopath.AbsoluteSystemPath, files []turbopath.AnchoredSystemPath, existing bool) (map[turbopath.AnchoredUnixPath]string, error) {
+	hashObject := make(map[turbopath.AnchoredUnixPath]string, len(files))
 	for _, file := range files {
 		hash, err := fs.GitLikeHashFile(file.RestoreAnchor(rootPath))
+		if errors.Is(err, &os.PathError{}) {
```

Comment:
Is there supposed to be an `existing &&` check here?

Reply:
Nah, `nil` can't ever be the thing; and it's type-constrained to `error`. Therefore `errors.Is` already covers the existence check.

Reply:
Gotcha. But are we using `existing` anywhere in this function then?

Reply:
Oh, also no! `GitLikeHashFile` calls `Open` as the first step, which blows up with an `os.PathError` if it fails.

So we simply try to read it and if that fails we inspect the error to make sure it's the one that you get if it doesn't exist.

A separate existence check would be an unnecessary bonus round trip to the FS.

Reply:
OH OH OH; yes there is a bug here. Test incoming. Thank you for your detailed review!